### PR TITLE
chore: Adding Rarity field for My Collection Add work form 

### PR DESCRIPTION
**Pull request by rajsam003**
The type of this PR is: Feature

This PR resolves [CX-2201]

### Description

This PR is the correction of [closed PR](https://github.com/artsy/eigen/pull/5947)

image

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Added rarity field on My Collection Artwork form - Sam

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[CX-2201]: https://artsyproduct.atlassian.net/browse/CX-2201?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Added rarity field on My Collection Artwork form - Sam)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 22c63c1de8225ee1e166679a7f37d967c34ba12c
</p>

File: src/lib/Scenes/MyCollection/Screens/ArtworkForm/Components/Rarity.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+              <Input
+                keyboardType="decimal-pad"
+                onChangeText={formik.handleChange("editionSize")}
+                onBlur={formik.handleBlur("editionSize")}
+                value={formik.values.editionSize}
+              />
+            </Flex>
+          </Flex>
+        </>
+      ) : null}
```
Comment (olerichter00) [inline src/lib/Scenes/MyCollection/Screens/ArtworkForm/Components/Rarity.tsx]:
```
      {formik.values.attributionClass === "LIMITED_EDITION" && (
        <Flex flexDirection="row" mt="2" alignItems="flex-end">
          <Flex flex={1}>
            <Flex>
              <Input
                title="Edition number"
                keyboardType="decimal-pad"
                onChangeText={formik.handleChange("editionNumber")}
                onBlur={formik.handleBlur("editionNumber")}
                value={formik.values.editionNumber}
              />
            </Flex>
          </Flex>
          <Flex justifyContent="center" mx={2} height={INPUT_HEIGHT}>
            <Text>/</Text>
          </Flex>
          <Flex flex={1}>
            <Flex>
              <Input
                title="Edition size"
                keyboardType="decimal-pad"
                onChangeText={formik.handleChange("editionSize")}
                onBlur={formik.handleBlur("editionSize")}
                value={formik.values.editionSize}
              />
            </Flex>
          </Flex>
        </Flex>
      )}
```

**Review by olerichter00 — APPROVED** [review]
🚀

File: src/lib/Scenes/MyCollection/Screens/ArtworkForm/Components/Rarity.tsx [hunk 2]
```diff
@@ -24,42 +24,35 @@ export const Rarity: React.FC = () => {
         options={artworkRarityClassifications}
         testID="rarity-select"
       />
-      {formik.values.attributionClass === "LIMITED_EDITION" ? (
-        <>
```
Comment (rajsam003) [inline src/lib/Scenes/MyCollection/Screens/ArtworkForm/Components/Rarity.tsx]: Sorry I didnt notice it before. What is the reason behind removing this? 👀

Comment (olerichter00) [reply src/lib/Scenes/MyCollection/Screens/ArtworkForm/Components/Rarity.tsx]: The titles were not aligned with the input components so I tweaked the layout a bit.

Comment (rajsam003) [reply src/lib/Scenes/MyCollection/Screens/ArtworkForm/Components/Rarity.tsx]: You added this `alignItems="flex-end"`. Incredible 🙌

Comment (olerichter00) [reply src/lib/Scenes/MyCollection/Screens/ArtworkForm/Components/Rarity.tsx]: Oh and PR is a feature and not a chore (see title).

**Pull request merged**